include/CL/sycl/accessor.hpp

```diff
@@ -589,14 +590,35 @@ accessMode == access::mode::discard_read_write) && dimensions == 1) */
 
 
   /* Available only when: accessMode == access::mode::atomic && dimensions == 0*/
-  //operator atomic<dataT, access::address_space::global_space> () const;
+  template<int D = dimensions,
+           access::mode M = accessmode,
+           typename = std::enable_if_t<M == access::mode::atomic && D == 0>>
+  HIPSYCL_UNIVERSAL_TARGET
+  operator atomic<dataT, access::address_space::global_space> () const
+  {
+    return atomic<dataT, access::address_space::global_space> { global_ptr<dataT>(_ptr) };
+  }
 
   /* Available only when: accessMode == access::mode::atomic && dimensions > 0*/
-  //atomic<dataT, access::address_space::global_space> operator[](
-  //    id<dimensions> index) const;
+  template<int D = dimensions,
+           access::mode M = accessmode,
+           typename = std::enable_if_t<(D > 0) && (M == access::mode::atomic)>>
+  HIPSYCL_UNIVERSAL_TARGET
+  atomic<dataT, access::address_space::global_space> operator[](
+      id<dimensions> index) const
+  {
+    return atomic<dataT, access::address_space::global_space>
+             { global_ptr<dataT>(_ptr + detail::linear_id<dimensions>::get(index, _buffer_range)) };
+  }
 
-  //atomic<dataT, access::address_space::global_space> operator[](
-  //    size_t index) const;
+  template<int D = dimensions,
+           access::mode M = accessmode,
+           typename = std::enable_if_t<(D == 0) && (M == access::mode::atomic)>>
```

Comment:
After thinking about a bit, I think there's a typo here. I think this should be enabled if `D==1` and not if `D==0`, because indexing 0-dimensional data doesn't make any sense.

Reply:
Good catch. And the other comparison should be `D > 1`. This should be fixed now.

Reply:
Why `D>1` for the other overload? A SYCL id can be one-dimensional, so I think `D>0` was correct. Am I missing something?

Reply:
Ah, good point. Fixed.